beacon-chain/rpc/beaconv1/pool.go

```diff
@@ -60,6 +85,19 @@ func (bs *Server) SubmitAttestations(ctx context.Context, req *ethpb.SubmitAttes
 			codes.Internal,
 			"Could not publish one or more attestations. Some attestations could be published successfully.")
 	}
+
+	if len(attFailures) > 0 {
+		failuresContainer := &attestationsVerificationFailure{Failures: attFailures}
+		jsonFailures, err := json.Marshal(failuresContainer)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not prepare attestation failure information: %v", err)
+		}
+		if err := grpc.SetHeader(ctx, metadata.Pairs(grpcutils.CustomErrorMetadataKey, string(jsonFailures))); err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not prepare attestation failure information: %v", err)
+		}
+		return nil, status.Errorf(codes.Internal, "One or more attestations failed validation")
```

Comment:
this stuff seems like a candidate for a helper function or package or middleware. It shouldn't be part of the basic logic of an endpoint

Reply:
Agreed, I will change this.